src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -120,21 +171,21 @@ public class AadIssuerValidator
 
             try
             {
-                if (securityToken.Issuer.EndsWith("v2.0", StringComparison.OrdinalIgnoreCase))
+                var effectiveConfigurationManager = GetEffectiveConfigurationManager();
+                string aadIssuer = effectiveConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult().Issuer;
+                if (IsValidIssuer(aadIssuer, tenantId, issuer))
                 {
-                    if (AadIssuerV2 == null)
-                        AadIssuerV2 = CreateConfigManager(AadAuthorityV2).GetConfigurationAsync().ConfigureAwait(false).GetAwaiter().GetResult().Issuer;
-
-                    if (IsValidIssuer(AadIssuerV2, tenantId, issuer))
-                        return issuer;
+                    // todo: set LKG only token is valid
+                    effectiveConfigurationManager.LastKnownGoodConfiguration = new OpenIdConnectConfiguration() { Issuer = aadIssuer };
+                    return issuer;
                 }
-                else
+                else if (validationParameters.ValidateWithLKG && effectiveConfigurationManager.LastKnownGoodConfiguration != null)
                 {
-                    if (AadIssuerV1 == null)
-                        AadIssuerV1 = CreateConfigManager(AadAuthorityV1).GetConfigurationAsync().ConfigureAwait(false).GetAwaiter().GetResult().Issuer;
-
-                    if (IsValidIssuer(AadIssuerV1, tenantId, issuer))
-                        return issuer;
+                    aadIssuer = effectiveConfigurationManager.LastKnownGoodConfiguration.Issuer;
+                    if (IsValidIssuer(aadIssuer, tenantId, issuer))
+                        return aadIssuer;
+                    else
+                        effectiveConfigurationManager.RequestRefresh();
```

Comment:
RequestRefresh here if validate with LKG failed

Reply:
Why not try GetConfiguration again after refresh, like in JsonWebTokenHandler?

Reply:
I request refresh the metadata here if validate with LKG failed and the handler will validate token with refreshed metadata for the third try.

Reply:
Added RefeshBeforeValidating flag